### PR TITLE
Equipment

### PR DESCRIPTION
**Pull request by Dabzelos**
simple logic of equipped items

### CODE REVIEW

**Review by Lesh79 — COMMENTED** [review]
You can also add the weight of each item so that the character can only carry N weights on himself

File: GameLogic/player_management/Equipment.cs [hunk 1]
```diff
@@ -8,16 +9,32 @@ public enum WeaponType
 }
 
 public class Helmet :Item{
+
+    public Helmet ( string name, int price, int def, double defPerc) : base(name, price)
+    {
+        Def = def;
+        DefPerc = defPerc;
+    }
```
Comment (Lesh79) [inline GameLogic/player_management/Equipment.cs:17]: Can't this be added to an abstract class? You write it everywhere.

File: GameLogic/player_management/Equipment.cs [hunk 2]
```diff
@@ -8,16 +9,32 @@ public enum WeaponType
...
     public int Def { set; get; }
     public double DefPerc { set; get; }
 }
 
 public class Chestplate :Item{
+    public Chestplate ( string name, int price, int def, double defPerc) : base(name, price)
+    {
+        Def = def;
+        DefPerc = defPerc;
+    }
```
Comment (Lesh79) [inline GameLogic/player_management/Equipment.cs:27]: Similarly

File: GameLogic/player_management/Equipment.cs [hunk 3]
```diff
@@ -8,16 +9,32 @@ public enum WeaponType
...
     public double DefPerc { set; get; }
 }
 
 public class Leggins :Item{
+    public Leggins ( string name, int price, int def, double defPerc) : base(name, price)
+    {
+        Def = def;
+        DefPerc = defPerc;
+    }
     public int Def { set; get; }
```
Comment (Lesh79) [inline GameLogic/player_management/Equipment.cs:38]: Similarly

File: GameLogic/player_management/item.cs [hunk 4]
```diff
@@ -4,6 +4,11 @@ namespace Chat_Warriors.GameLogic.player_management;
 [Keyless]
 public class Item
 {
+    public Item(string name, int price){ //конструктор айтема
```
Comment (Lesh79) [inline GameLogic/player_management/item.cs:7]: Don't you think you need to make this class abstract?

**Review by Lesh79 — COMMENTED** [review]
Please change the get and set places again, as for me, it would be better

File: GameLogic/player_management/Equipment.cs (hunk 1 above)
Comment (Dabzelos) [reply GameLogic/player_management/Equipment.cs:17]: Nah, cause I have class Armour(I think about renaming it to armourSet) and it must contain three pieces of Set, each with specified type: helmet, chest plate and leggins, 
Refactoring will cause some problems checking two items of armor of the same type in a set (for example, two helmets)

File: GameLogic/player_management/Equipment.cs [hunk 5]
```diff
@@ -59,13 +58,27 @@ public Armour(Chestplate? chest = null, Leggins? legs = null, Helmet? helmet = n
...
+    public Helmet Helm { get; set; }
+    public Leggins Leggins { get; set; }
 
     public int TotalDefence()
     {
-        return Chest.Def + Leggins.Def + Helm.Def;
+        int chestDef = 0;
+        int helmDef = 0;
+        int leggDef = 0;
+        if (Chest != null)
```
Comment (Lesh79) [inline GameLogic/player_management/Equipment.cs:70]: The ternary operator...

File: GameLogic/player_management/Equipment.cs [hunk 6]
```diff
@@ -75,16 +88,26 @@ public class Weapon: Item
...
+    public int Attack { get; set; }
 }
 
 public class Equipment
 {
-    public Armour EquippedArmour(Chestplate chest = null, Leggins legs = null, Helmet helmet = null)
+    public Equipment()
     {
-        return new Armour(chest, legs, helmet);
+        EquipArmour();
```
Comment (Lesh79) [inline GameLogic/player_management/Equipment.cs:99]: ?

**Comment by Lesh79** [comment]
Perhaps you can use the "Abstract Factory" pattern, but so far it only seems to me. Read and think about it

File: GameLogic/player_management/Equipment.cs [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    public int TotalDefence()
+    {
+        return Chest.Def + Leggins.Def + Helm.Def;
+    }
+}
+
+public class Weapon: Item
+{
+    public WeaponType WeaponType { set; get; }
+    public int Attack { set; get; }
```
[the remark this answers is not in the document]
Comment (Dabzelos) [reply GameLogic/player_management/Equipment.cs]: Kinda yes, but I thought about some modifications of attack force
Like critchance, so I need basic attack of item then, in Player.totalattack we will see the total possible damage
I'll push these changes when I get home

File: GameLogic/player_management/Equipment.cs (hunk 6 above)
Comment (Dabzelos) [reply GameLogic/player_management/Equipment.cs:99]: after initializing a player we initialize its armour but without any elements

File: GameLogic/player_management/Equipment.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        if (legs != null)
+        {
+            Leggins = legs;
+        }
+
+        if (helmet != null)
+        {
+            Helm = helmet;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Dabzelos) [reply GameLogic/player_management/Equipment.cs:59]: no.

File: GameLogic/player_management/Equipment.cs (hunk 6 above)
Comment (Lesh79) [reply GameLogic/player_management/Equipment.cs:100]: Should it be in the constructor of this class? Is it better to make the method static?

**Pull request closed without merging**